automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/apps/AppCreationSpec.scala

```diff
@@ -79,17 +79,20 @@ class AppCreationSpec extends GPAllocFixtureSpec with LeonardoTestUtils with GPA
             120,
             10 seconds
           ).compile.lastOrError
+          // TODO remove first case in below if statement when Galaxy deletion is reliable
           _ <- if (!deletedDoneCheckable.isDone(monitorDeleteResult)) {
             loggerIO.warn(
               s"AppCreationSpec: app ${googleProject.value}/${appName.value} did not finish deleting after 20 minutes. Result: $monitorDeleteResult"
             )
           } else {
-            loggerIO.info(
-              s"AppCreationSpec: app ${googleProject.value}/${appName.value} delete result: $monitorDeleteResult"
-            )
+            // Verify creating another app with the same disk doesn't error out
+            for {
+              _ <- loggerIO.info(
+                s"AppCreationSpec: app ${googleProject.value}/${appName.value} delete result: $monitorDeleteResult"
+              )
+              _ <- LeonardoApiClient.createAppWithWait(googleProject, restoreAppName, createAppRequest)
+            } yield ()
           }
-//          // Verify creating another app with the same disk doesn't error out
-//          _ <- LeonardoApiClient.createAppWithWait(googleProject, restoreAppName, createAppRequest)
```

Comment:
I think may be why we were seeing "namespace already exists".. if the app deletion fails due to the flakey Galaxy pre-delete job, then this test would _still_ go on to create the second app while the namespace still exists in k8s.

Reply:
yea..that makes sense....should've read the existing test code a bit closer :)